### ISSUE TITLE
Full Text Search - WordPress database error: Invalid object name

### ISSUE THREAD

**Issue by Vidocq24601** [ProjectNami/projectnami#264]
Multisite PN Installation issue with WordPress search widget. 

php_errors.log contains multiple errors following this format when WP search widget is attempted to be used:

[15-Feb-2018 19:07:24 UTC] WordPress database error [Microsoft][ODBC Driver 13 for SQL Server][SQL Server]Invalid object name 'wp_37_fulltext_search'. Code - 208 for query SELECT  wp_37_posts.* , wp_37_posts.post_date FROM wp_37_posts  INNER JOIN CONTAINSTABLE(wp_37_fulltext_search, *, 'tour') as pnftsearch on pnftsearch.[KEY] = wp_37_posts.ID  WHERE 1=1  AND wp_37_posts.post_type IN ('post', 'page', 'attachment') AND (wp_37_posts.post_status = 'publish')  ORDER BY  pnftsearch.[RANK] desc  OFFSET  0 ROWS FETCH NEXT 10 ROWS ONLY made by require('D:\home\site\wwwroot\wp-blog-header.php'), wp, WP->main, WP->query_posts, WP_Query->query, WP_Query->get_posts
[15-Feb-2018 23:10:08 UTC] PHP Warning:  sqlsrv_query() expects parameter 1 to be resource, null given in D:\home\site\wwwroot\wp-includes\wp-db.php on line 1879

Additional Information:

1) Error occurs across subsites and themes on this WebApp. Widget search operates as expected when Full Text Search Plugin is deactivated.

2) On separate PN Multisite installation, network activated full text search plugin is working with WP search widget as expected.

The impacted WebApp was deployed from Azure Gallery ~1 year ago, only recently noticed errors in logs, but this may have been an unnoticed issue as search function is not often utilized. Have attempted manually updating Full Text Search Plugin, but still generating the same 'Invalid object name' error in log.

Any idea on what to research as far as what the issue is? Other than search, sites are currently operating as expected without other problems.

**Comment by patrickebates** [comment]
Please try to execute the following SQL statement against the DB, and provide the results...

`CREATE VIEW [dbo].[wp_37_fulltext_search] WITH SCHEMABINDING AS select wp_37_posts.ID, display_name + ' ' + post_title + ' ' + post_excerpt + ' ' + post_content as search_text from dbo.wp_37_posts inner join dbo.wp_users on wp_37_posts.post_author = wp_users.ID`

**Comment by Vidocq24601** [comment]
"Query succeeded: Affected rows: 0."

Then attempting to use WP Search Widget, new error:

[16-Feb-2018 18:25:08 UTC] WordPress database error [Microsoft][ODBC Driver 13 for SQL Server][SQL Server]Cannot use a CONTAINS or FREETEXT predicate on table or indexed view 'wp_37_fulltext_search' because it is not full-text indexed. Code - 7601 for query SELECT COUNT(  wp_37_posts.ID ) as [found_rows] FROM wp_37_posts  INNER JOIN CONTAINSTABLE(wp_37_fulltext_search, *, 'tour') as pnftsearch on pnftsearch.[KEY] = wp_37_posts.ID  WHERE 1=1  AND wp_37_posts.post_type IN ('post', 'page', 'attachment') AND (wp_37_posts.post_status = 'publish')  made by require('D:\home\site\wwwroot\wp-blog-header.php'), wp, WP->main, WP->query_posts, WP_Query->query, WP_Query->get_posts

[16-Feb-2018 18:25:08 UTC] WordPress database error [Microsoft][ODBC Driver 13 for SQL Server][SQL Server]Cannot use a CONTAINS or FREETEXT predicate on table or indexed view 'wp_37_fulltext_search' because it is not full-text indexed. Code - 7601 for query SELECT  wp_37_posts.* , wp_37_posts.post_date FROM wp_37_posts  INNER JOIN CONTAINSTABLE(wp_37_fulltext_search, *, 'tour') as pnftsearch on pnftsearch.[KEY] = wp_37_posts.ID  WHERE 1=1  AND wp_37_posts.post_type IN ('post', 'page', 'attachment') AND (wp_37_posts.post_status = 'publish')  ORDER BY  pnftsearch.[RANK] desc  OFFSET  0 ROWS FETCH NEXT 10 ROWS ONLY made by require('D:\home\site\wwwroot\wp-blog-header.php'), wp, WP->main, WP->query_posts, WP_Query->query, WP_Query->get_posts

**Comment by patrickebates** [comment]
ok, now execute each of the following commands in sequence...

```
if not exists (select * from sys.dm_fts_active_catalogs where name = 'ftCatalog') create fulltext catalog ftCatalog

CREATE UNIQUE CLUSTERED INDEX [CLU_wp_37_fulltext_search] ON [dbo].[wp_37_fulltext_search] ([ID] ASC)

CREATE FULLTEXT INDEX ON wp_37_fulltext_search(search_text) KEY INDEX CLU_wp_37_fulltext_search ON ftCatalog

ALTER FULLTEXT INDEX ON wp_37_fulltext_search ENABLE

```

**Comment by Vidocq24601** [comment]
Ok great, search is now working on that subsite!

**Important question though:**

Search is still not working on the other subsites. Does that mean I will need to execute the original SQL statement against the DB and also the most recent commands in that order for every subsite on the network (replacing "37" with each individual site number)? 

There are a lot of sites on the network and more that are going to be added. If there isn't a better way, I can of course do that for each subsite, but I would also really love to understand what the issue was. 

Thanks so much for your prompt follow up and troubleshooting.

**Comment by patrickebates** [comment]
These commands should have fired as part of the plugin activation for each site.  So I'm curious why they didn't execute since you have no errors.

By any chance is the search plugin Network Activated?  I so, I wonder if that is preventing it from executing properly with each new site.  The solution might be to enable it on each site individually.

**Comment by Vidocq24601** [comment]
Patrick, you are correct. The plugin was Network Activated. I network deactivated and activated individually on another subsite and then WP search function worked as expected. 

Additionally, I was able to recreate this scenario on the other WebApp. Search was only operating on the main site and not the subsites. 

Thanks again for your help in clearing up the issue. Marking as closed since we established it was network activation problem and the solution is simply individually activating plugin on subsites.
